pkg/provider/github/github.go

```diff
@@ -543,3 +543,33 @@ func uniqueRepositoryID(repoIDs []int64, id int64) []int64 {
 	}
 	return r
 }
+
+// listBranches list all branches in the repo and set the default branch for a matching branch commit SHA.
+func (v *Provider) listBranches(ctx context.Context, runevent *info.Event) error {
```

Comment:
what happen if the commit is in multiple branches?

Reply:
I believe commit sha will be unique :thinking: